tlv/stream.go

```diff
@@ -202,8 +219,8 @@ func (s *Stream) decode(r io.Reader, parsedTypes TypeMap) (TypeMap, error) {
 		// Place a soft limit on the size of a sane record, which
 		// prevents malicious encoders from causing us to allocate an
 		// unbounded amount of memory when decoding variable-sized
-		// fields.
-		if length > MaxRecordSize {
+		// fields. This is only checked when the p2p bool is true.
+		if p2p && length > MaxRecordSize {
```

Comment:
Just a thought. Could `p2p` live in the `Stream` so it doesn't need to get passed in as a function parameter?